src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCacheCursor.cs

```diff
@@ -83,16 +86,22 @@ public virtual IBatchContainer GetCurrent(out Exception exception)
         /// <returns></returns>
         public virtual bool MoveNext()
         {
+            if (current == null && IsSet)
+            {
+                current = Element.Value.Batch;
```

Comment:
Do we know this item is in the stream?

Reply:
Ah, good point. Nothing particularly tricky about fixing that, thankfully, (literally adding `&& IsInStream(Element.Value.Batch)` to the if statement handles that.